### ISSUE TITLE
CI BUG: it probably shouldn't run on branches of forks

### ISSUE THREAD

**Issue by bsipocz** [networkx/networkx#7504]
CI has been triggered on my fork while opening #7503 which is very much not standard behaviour. Is that intentional, or the workflows need an audit to ensure they are not triggered on forks?

**Comment by rossbar** [comment]
Yes, most actions triggered on push/PR run on forks. I'm not sure what the best practice is here - I had always assumed that when jobs were enabled on forks then GH was smart enough to use the results from the fork's actions to populate the CI report in the base repo. I've never actually checked this though - one would think GH would enforce such behavior, otherwise there are probably huge amounts of duplicate CI jobs being run on the platform.

FWIW I sometimes rely on CI on my fork to make sure something will work on all platforms (e.g. windows) before I open a PR; so I'm not sure that disabling CI on forks across the board is necessarily something we'd want to do (scheduled jobs are already disabled on forks). WDYT @bsipocz , are additional pros/cons that come to mind?

**Comment by RohitP2005** [comment]
Hey @rossbar ,  

I'd like to work on this issue. My proposal:  
- Restrict CI to PRs targeting the main repo.  
- Allow manual CI runs on forks if needed.  
- Optionally, keep minimal CI checks on forks.  

Let me know your thoughts! 🚀

**Comment by rossbar** [comment]
IMO this is more of a discussion than something that needs implementation. In particular, I'd be interested in learning:
 1. How github actions handles CI jobs across forks. Particularly, when a job is triggered by a push to a fork, is that action reflected in the PR or is it duplicated (i.e. there is an independent job running on both the fork *and* the main repo)
 2. What the best-practices are for other open-source libraries.

So if you're interested in researching around a bit, by all means!

**Comment by bsipocz** [comment]
Yeap, it's a duplicated run onCe the PR is in place, thus I was opening the issue the first place. 
I haven't really seen this done elsewhere, but that's said, it can be a totally valid approach if the expectation here is to open practically finished PRs (in a couple of my projects I'm doing the opposite and encourage prs very early on, so I don't end up reviewing huge PRs that the authors see as finished and ready to go)

**Comment by bsipocz** [comment]
Otherwise I agree, this is more like a discussion and a decision for the maintainers rather than a "help wanted" type issue.

**Comment by rossbar** [comment]
We looked into this a bit in the community meeting and found that it seems to be under the user's fork settings. For example, on your fork if you go to `Settings` then `Actions` in the left panel, then `General` - you get to a page that gives 4 options relating to action configuration. The first two of these are "Allow all actions and reusable workflows" and "Disable actions".

To summarize - I think we're then in the state we want to be: users can control whether or not actions run on their fork.

**Comment by MridulS** [comment]
> To summarize - I think we're then in the state we want to be: users can control whether or not actions run on their fork.

I'll go ahead and close this!
